scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroup.scala

```diff
@@ -309,6 +310,53 @@ object DynamicTLSPeerGroup {
     */
   case class IncomingConnectionThrottlingConfig(throttleLocalhost: Boolean, throttlingDuration: FiniteDuration)
 
+  /**
+    * Configures detection on idle peer
+    * @param timeUnit unit of time for all time configs
+    * @param readerIdleTime a ChannelIdle event whose state is ReaderIdle will be triggered when
+    *                       no read was performed for the specified period of time on given channel. Specify 0 to disable.
+    * @param writerIdleTime a ChannelIdle event whose state is WriterIdle will be triggered when
+    *                       no write was performed for the specified period of time on given channel. Specify 0 to disable.
+    * @param allIdleTime a ChannelIdle event whose state is AllIdle will be triggered when
+    *                       no read was performed for the specified period of time on given channel. Specify 0 to disable.
+    *
+    */
+  class StalePeerDetectionConfig private (
+      val timeUnit: TimeUnit,
+      val readerIdleTime: Long,
+      val writerIdleTime: Long,
+      val allIdleTime: Long
+  )
+
+  object StalePeerDetectionConfig {
+    def apply(
+        timeUnit: TimeUnit,
+        readerIdleTime: Long,
+        writerIdleTime: Long,
+        allIdleTime: Long
```

Comment:
Why not use `FiniteDuration`? You can use whatever time unit you want inside, but the chance of misconfiguration would be lower, e.g. starting with `HOUR,1,1,1` then modifying it to `MINUTE,10,1,1` or something.

Reply:
I thought about it but ultimately this conveys intent that all things should have the same unit better, also using finite duration is error prone it way that user of library can pass `5.second` in one field, and `5.days` in other. 

One thing that is in favour of finite duration is that it is more scala friendly api, but I am not sure that is enough advantage.

Reply:
`FiniteDuration` also works with the configuration. I think if the user configures 5 days, that's their problem, it's easy enough to read, and for us to find a common denominator.

Reply:
`FiniteDuration` also works with the configuration. I think if the user configures 5 days, that's their problem, it's easy enough to read, and for us to find a common denominator.

Reply:
I am still not fully convinced, but maybe having `FinititeDuration` for the sake of easy reading configs from HOCON seems neat. lets go this way then 👍